core/src/main/java/org/infinispan/statetransfer/StateConsumerImpl.java

```diff
@@ -457,18 +477,19 @@ public void onTopologyUpdate(final CacheTopology cacheTopology, final boolean is
                localTopologyManager.confirmRebalancePhase(cacheName, cacheTopology.getTopologyId(), cacheTopology.getRebalanceId(), null);
          }
 
+         PartitionHandlingConfiguration phConfig = configuration.clustering().partitionHandling();
+         boolean wasMember = previousWriteCh != null && previousWriteCh.getMembers().contains(rpcManager.getAddress());
+         boolean deletePastMemberVals = wasMember && phConfig.whenSplit() != PartitionHandling.ALLOW_READ_WRITES && phConfig.mergePolicy() == null;
```

Comment:
OK, I've mistaken that for MergePolicy.NONE in parser. Shouldn't this be the default to keep the behaviour backwards compatible in minor release, then?